pkgs/development/tools/bpf-linker/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, llvmPackages_14
+, zlib
+, ncurses
+, libxml2
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "bpf-linker";
+  version = "0.9.4";
+
+  src = fetchFromGitHub {
+    owner = "aya-rs";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-jYuBk78aGQLUeNF6d6kjGPuMxEF22XJquHcs23WVGm0=";
+  };
+
+  cargoHash = "sha256-X8EVpOxDHwE/wj/gly/wdZ6tsrMrz3kkDe9gEPbk6iw=";
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [ "system-llvm" ];
+
+  nativeBuildInputs = [ llvmPackages_14.llvm ];
+  buildInputs = [ zlib ncurses libxml2 ];
+
+  # fails with: couldn't find crate `core` with expected target triple bpfel-unknown-none
+  # rust-src and `-Z build-std=core` are required to properly run the tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Simple BPF static linker";
+    homepage = "https://github.com/aya-rs/bpf-linker";
+    license = with licenses; [ asl20 mit ];
+    maintainers = with maintainers; [ nickcao ];
+    # llvm-sys crate locates llvm by calling llvm-config
+    # which is not available when cross compiling
```

Comment:
This doesn't make sense to me. LLVM is in nativeBuildInputs, so it would be available when cross compiling, no?

Reply:
LLVM should actually be in `buildInputs`, having it in `nativeBuildInputs` is a hack for making `llvm-sys` happy, while breaking cross compilation.

Reply:
Looking at `llvm-sys` and its build script now. They say that cross-compilation should theoretically work (but they haven't tested it), yet I don't see how it could work considering that they run the `llvm-config` binary...

Reply:
It could work if we provide a wrapper script for llvm-config, which in turn calls llvm-config-native instead, but I don't think that worth the effort.